question/bank/bulkmove/move.php

```diff
@@ -0,0 +1,124 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Move questions page.
+ *
+ * @package    qbank_bulkmove
+ * @copyright  2021 Catalyst IT Australia Pty Ltd
+ * @author     Safat Shahin <safatshahin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once(__DIR__ . '/../../../config.php');
+require_once(__DIR__ . '/../../editlib.php');
+global $DB, $OUTPUT, $PAGE, $COURSE;
+
+$moveselected = optional_param('move', false, PARAM_BOOL);
+$returnurl = optional_param('returnurl', 0, PARAM_LOCALURL);
+$cmid = optional_param('cmid', 0, PARAM_INT);
+$courseid = optional_param('courseid', 0, PARAM_INT);
+$category = optional_param('category', null,PARAM_SEQUENCE);
+
+if ($returnurl) {
+    $returnurl = new moodle_url($returnurl);
+}
+
+\core_question\local\bank\helper::require_plugin_enabled('qbank_bulkmove');
+
+if ($cmid) {
+    list($module, $cm) = get_module_from_cmid($cmid);
+    require_login($cm->course, false, $cm);
+    $thiscontext = context_module::instance($cmid);
+} else if ($courseid) {
+    require_login($courseid, false);
+    $thiscontext = context_course::instance($courseid);
+} else {
+    throw new moodle_exception('missingcourseorcmid', 'question');
+}
+
+$contexts = new question_edit_contexts($thiscontext);
+$url = new moodle_url('/question/bank/bulkmove/move.php');
+
+$PAGE->set_url($url);
+$streditingquestions = get_string('movequestions', 'qbank_bulkmove');
+$PAGE->set_title($streditingquestions);
+$PAGE->set_heading($COURSE->fullname);
+
+if ($category) {
+    list($tocategoryid, $contextid) = explode(',', $category);
+    if (! $tocategory = $DB->get_record('question_categories',
+        ['id' => $tocategoryid, 'contextid' => $contextid])) {
+        throw new \moodle_exception('cannotfindcate', 'question');
+    }
+}
+
+if ($moveselected && ($confirm = optional_param('confirm', '', PARAM_ALPHANUM))) {
+    $moveselected = required_param('moveselected', PARAM_RAW);
+    if ($confirm == md5($moveselected)) {
+        if ($questionids = explode(',', $moveselected)) {
+            list($usql, $params) = $DB->get_in_or_equal($questionids);
+            $questions = $DB->get_records_sql("
+                        SELECT q.*, c.contextid
+                        FROM {question} q
+                        JOIN {question_categories} c ON c.id = q.category
+                        WHERE q.id {$usql}", $params);
+            foreach ($questions as $question) {
+                question_require_capability_on($question, 'move');
+            }
+            question_move_questions_to_category($questionids, $tocategory->id);
+        }
+    }
+    redirect(new moodle_url($returnurl, ['category' => "{$tocategoryid},{$contextid}"]));
+}
+
+
+echo $OUTPUT->header();
+
+if ($moveselected) {
+    $rawquestions = $_REQUEST;
```

Comment:
I'm not sure we should be accessing the super global directly, but I can't find any docs to say we can't